sbi/neural_nets/density_estimators/categorical_net.py

```diff
@@ -176,11 +176,11 @@ def loss(self, input: Tensor, condition: Tensor, **kwargs) -> Tensor:
         r"""Return the loss for training the density estimator.
 
         Args:
-            input: Inputs of shape `(sample_dim, batch_dim, *input_event_shape)`.
+            input: Inputs of shape `(batch_dim, *input_event_shape)`.
             condition: Conditions of shape `(batch_dim, *condition_event_shape)`.
 
         Returns:
             Loss of shape `(batch_dim,)`
         """
 
-        return -self.log_prob(input, condition)
+        return -self.log_prob(input.unsqueeze(0), condition)[0]
```

Comment:
why is this now needed?

Reply:
`.log_prob` takes inputs of shape `(sample_dim, batch_dim, event)`, but `loss` now takes `(batch_dim, event)`. That's why we have to unsqueeze.

Similarly, `log_prob` returns `(sample_dim, batch_dim)`, but `.loss` is supposed to return `(batch_dim)`.